pkg/cmd/pkg/package_install.go

```diff
@@ -1,28 +1,163 @@
+	}
+	pkg := description.Package
+	link := "https://mesosphere.com/catalog-terms-conditions/#certified-services"
+	if !pkg.Selected {
+		fmt.Fprint(ctx.ErrOut(), "This is a Community service. "+
+			"Community services are not tested for production environments. "+
+			"There may be bugs, incomplete features, incorrect documentation, or other discrepancies.\n")
+		link = "https://mesosphere.com/catalog-terms-conditions/#community-services"
+	}
+	fmt.Fprintf(ctx.ErrOut(), "By Deploying, you agree to the Terms and Conditions %s\n", link)
+	if opts.appOnly && pkg.PreInstallNotes != "" {
+		fmt.Fprintf(ctx.ErrOut(), "%s\n", pkg.PreInstallNotes)
+	}
+	if !opts.yes {
+		err = ctx.Prompt().Confirm("Continue installing? [yes/no] ", "Yes")
+		if err != nil {
+			return err
+		}
+	}
+	if opts.appOnly && description.Package.Marathon.V2AppMustacheTemplate != "" {
+		fmt.Fprintf(ctx.ErrOut(), "Installing Marathon app for package [%s] version [%s]\n", packageName, pkg.Version)
+		err := c.PackageInstall(opts.appID, packageName, pkg.Version, opts.optionsPath)
+		if err != nil {
+			return err
+		}
+	}
+	if opts.cliOnly && isEmptyCli(pkg.Resource.Cli) && !isEmptyCommand(pkg.Command) {
+		return fmt.Errorf("unable to install CLI subcommand. PIP subcommands are no longer supported" +
+			" see: https://godoc.org/github.com/dcos/client-go/dcos#CosmosPackageCommand")
+	}
+	if opts.cliOnly && !isEmptyCli(pkg.Resource.Cli) {
+		fmt.Fprintf(ctx.ErrOut(), "Installing CLI subcommand for package [%s] version [%s]\n", packageName, pkg.Version)
+		pluginInfo, err := cosmos.CLIPluginInfo(description.Package, pluginutil.HTTPClient("").BaseURL())
+		if err != nil {
+			return fmt.Errorf("cannot get plugin info: %s", err)
+		}
+
+		var checksum plugin.Checksum
+		for _, contentHash := range pluginInfo.ContentHash {
+			switch contentHash.Algo {
+			case dcos.SHA256:
+				checksum.Hasher = sha256.New()
+				checksum.Value = contentHash.Value
+			default:
+				fmt.Fprintf(ctx.ErrOut(), "unknown algorithm: %s\n", contentHash.Algo)
+			}
+		}
+
+		cluster, err := ctx.Cluster()
+		if err != nil {
+			return fmt.Errorf("cannot get cluster: %s", err)
+		}
+
+		plugin, err := ctx.PluginManager(cluster).Install(pluginInfo.Url, &plugin.InstallOpts{
+			Name:     packageName,
+			Update:   true,
+			Checksum: checksum,
+			PostInstall: func(fs afero.Fs, pluginDir string) error {
+				pkgInfoFilepath := filepath.Join(pluginDir, "package.json")
+				pkgInfoFile, err := fs.OpenFile(pkgInfoFilepath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+				if err != nil {
+					return err
+				}
+				defer pkgInfoFile.Close()
+				return json.NewEncoder(pkgInfoFile).Encode(description.Package)
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("cannot install plugin: %s", err)
+		}
+
+		plural := ""
+		if len(plugin.Commands) > 1 {
+			plural = "s"
+		}
+		cmds := make([]string, 0, len(plugin.Commands))
+		for _, c := range plugin.Commands {
+			cmds = append(cmds, c.Name)
+		}
+		sort.Strings(cmds)
```

Comment:
You could as well use the `plugin.CommandNames()` helper to get a sorted list of commands.

Reply:
Fixed